client/asset/btc/spv_peer_manager.go

```
@@ -0,0 +1,308 @@
+		}
+		return fmt.Errorf("failed to resolve address: %v", err)
+	}
+
+	if duplicatePeer, found := w.peerWithResolvedAddr(resolvedAddr); found {
+		return fmt.Errorf("%s and %s resolve to the same node", duplicatePeer, addr)
+	}
+
+	w.peers[addr] = &walletPeer{source: source, resolvedName: resolvedAddr}
+
+	if !initialLoad {
+		savedPeers, err := w.loadSavedPeersFromFile()
+		if err != nil {
+			w.log.Errorf("failed to load saved peers from file")
+		} else {
+			savedPeers[addr] = source
+			err = w.writeSavedPeersToFile(savedPeers)
+			if err != nil {
+				w.log.Errorf("failed to add peer to saved peers file: %v")
+			}
+		}
+	}
+
+	connectedPeers := w.connectedPeers()
+	_, connected := connectedPeers[resolvedAddr]
+	if !connected {
+		return w.cs.AddPeer(resolvedAddr)
+	}
+
+	return nil
+}
+
+// AddPeer connects to a new peer and stores it in the db.
+func (w *SPVPeerManager) AddPeer(addr string) error {
+	return w.addPeer(addr, added, false)
+}
+
+// RemovePeer disconnects from a peer added by the user and removes it from
+// the db.
+func (w *SPVPeerManager) RemovePeer(addr string) error {
+	w.peersMtx.Lock()
+	defer w.peersMtx.Unlock()
+
+	peer, found := w.peers[addr]
+	if !found {
+		return fmt.Errorf("peer not found: %v", addr)
+	}
+
+	savedPeers, err := w.loadSavedPeersFromFile()
+	if err != nil {
+		return err
+	}
+	delete(savedPeers, addr)
+	err = w.writeSavedPeersToFile(savedPeers)
+	if err != nil {
+		w.log.Errorf("failed to delete peer from saved peers file: %v")
+	} else {
+		delete(w.peers, addr)
+	}
+
+	connectedPeers := w.connectedPeers()
+	_, connected := connectedPeers[peer.resolvedName]
+	if connected {
+		return w.cs.RemoveNodeByAddr(peer.resolvedName)
+	}
+
+	return nil
+}
+
+// ConnectToInitialWalletPeers connects to the default peers and the peers
+// that were added by the user and persisted in the db.
+func (w *SPVPeerManager) ConnectToInitialWalletPeers() {
+	for _, peer := range w.defaultPeers {
+		w.addPeer(peer, defaultPeer, true)
```

Please comment on reasoning for ignoring error, if intended.